src/Common/Core/Impl/Services/PropertyDictionary.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Microsoft.Common.Core.Diagnostics;
+
+namespace Microsoft.Common.Core.Services {
+    /// <summary>
+    /// Allows property owners to control the lifetimes of the properties in the collection. 
+    /// </summary>
+    /// <remarks>This collection is synchronized in order to allow access by multiple threads.</remarks>
+    public sealed class PropertyDictionary {
```

Comment:
Not used anywhere just yet. It will be used in the editor space to implement `IPropertyOwner` that allows attaching things to buffer and view. Current code uses VS implementation but with portable code we'll need our own.